modules/vectra.py

```diff
@@ -522,6 +569,122 @@ def delete_rule(self, rule_id=None, restore_detections=True):
         return requests.delete('{url}/rules/{id}'.format(url=self.url, id=rule_id), headers=self.headers, params=params,
                                verify=self.verify)
 
+    @validate_api_v2
+    @request_error_handler
+    def get_group_by_id(self, group_id):
+        """
+        Get groups by id
+        :param rule_id: id of group to retrieve
+        """
+        return requests.get('{url}/groups/{id}'.format(url=self.url, id=group_id), headers=self.headers, verify=False)
+
+    @validate_api_v2
+    def get_all_groups(self):
+        """
+        Get all existing get_all_groups
+        Groups do not implement pagination
+        For consistency's sake, we still implement a generator
+        """
+        yield requests.get('{url}/groups'.format(url=self.url), headers=self.headers, verify=False)
```

Comment:
This endpoint should have pagination added, similar to the other endpoints, if it does, then we need to use 'next..

Reply:
But this is not the case now, right? I'd say we add pagination support on the API tools when the endpoint exposes it, no?

Reply:
Yeah we should update it in this PR get the groups from resp.json()['results'] and iterate through the pages similar to get_hosts.